### DISCUSSION TITLE
Storage API - Track Upload/Download Progress

### DISCUSSION THREAD

**Discussion by tommyc38** [supabase/supabase#1073] [Questions] [answered]
I didn't see it anywhere in the documentation but is there a way to track upload/download progress?

**Comment by heauton** [comment]
I was looking for this too. Perhaps API isn't there yet?

**Comment by inian** [comment, marked as answer]
Hi @tommyc38 and @heauton, yeah the backend doesn't support this yet. Not sure of an easy way to emit progress events from the backend off the top of my head. Creating an issue [here](https://github.com/supabase/storage-api/issues/23). Appreciate any ideas of implementing this :)

We will close this and mark it as answered!

**Comment by inian** [comment]
You can listen to progress events with our resumable uploads feature in [Storage v3](https://supabase.com/blog/storage-v3-resumable-uploads). Here are the docs on [resumable uploads](https://supabase.com/docs/guides/storage/uploads#resumable-upload).
